plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -533,137 +533,136 @@ class Pub(
         // Flag if the project is a Flutter project.
         var containsFlutter = false
 
-        listOf("packages"/*, "packages-dev"*/).forEach {
-            lockfile[it]?.fields()?.forEach { (packageName, pkgInfoFromLockfile) ->
-                try {
-                    val version = pkgInfoFromLockfile["version"].textValueOrEmpty()
-                    var description = ""
-                    var rawName = ""
-                    var homepageUrl = ""
-                    var vcs = VcsInfo.EMPTY
-                    var authors = emptySet<String>()
-
-                    val source = pkgInfoFromLockfile["source"].textValueOrEmpty()
-
-                    when {
-                        source == "path" -> {
-                            rawName = packageName
-                            val path = pkgInfoFromLockfile["description"]["path"].textValueOrEmpty()
-                            vcs = VersionControlSystem.forDirectory(workingDir.resolve(path))?.getInfo() ?: run {
-                                logger.warn {
-                                    "Invalid path of package $rawName: " +
-                                        "'$path' is outside of the project root '$workingDir'."
-                                }
-
-                                VcsInfo.EMPTY
+        lockfile.packages.forEach { (packageName, pkgInfoFromLockfile) ->
```

Comment:
The outer `listOf("packages"/*, "packages-dev"*/).forEach {` should be removed in a preceding commit as it's unrelated to this change and deserves a dedicated rationale. That also reduces nesting already beforehand so that the relevant changes here become more obvious.